apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java

```diff
@@ -249,13 +282,47 @@ protected Optional<TableMetadata> mapTableResponseToTableMetadata(
     return Optional.of(builder.build());
   }
 
+  protected List<TableDataLayoutMetadata> mapTableResponseToTableDataLayoutMetadataList(
+      GetTableResponseBody shallowResponseBody) {
+    GetTableResponseBody tableResponseBody =
+        getTable(shallowResponseBody.getDatabaseId(), shallowResponseBody.getTableId());
+
+    if (tableResponseBody == null) {
+      log.error(
+          "Error while fetching metadata for table: {}.{}",
+          shallowResponseBody.getDatabaseId(),
+          shallowResponseBody.getTableCreator());
+      return Collections.emptyList();
+    }
+
+    TableDataLayoutMetadata.TableDataLayoutMetadataBuilder<?, ?> builder =
+        TableDataLayoutMetadata.builder()
+            .creator(tableResponseBody.getTableCreator())
+            .dbName(tableResponseBody.getDatabaseId())
+            .tableName(tableResponseBody.getTableId())
+            .isPrimary(
+                tableResponseBody.getTableType()
+                    == GetTableResponseBody.TableTypeEnum.PRIMARY_TABLE)
+            .isTimePartitioned(tableResponseBody.getTimePartitioning() != null)
+            .isClustered(tableResponseBody.getClustering() != null)
+            .retentionConfig(getTableRetention(tableResponseBody).orElse(null));
+    if (tableResponseBody.getCreationTime() != null) {
+      builder.creationTimeMs(tableResponseBody.getCreationTime());
+    }
+    List<TableDataLayoutMetadata> result = new ArrayList<>();
+    for (DataLayoutStrategy strategy : getDataLayoutStrategies(tableResponseBody)) {
```

Comment:
Right now you're adding all per tableResponseBody, shouldn't it be per TableDataLayoutMetadata instead? Or you left it because ranking will get inserted here in the next PR?

Reply:
In line 314 we reuse builder as a template, and we set a new strategy and create a new TableDataLayoutMetadata object every time.